nf_core/modules/test_yml_builder.py

```diff
@@ -205,7 +205,11 @@ def create_test_file_dict(self, results_dir):
                 elem_md5 = self._md5(elem)
                 # Switch out the results directory path with the expected 'output' directory
                 elem = elem.replace(results_dir, "output")
-                test_files.append({"path": elem, "md5sum": elem_md5})
+                test_file = {"path": elem}
+                # Only add the md5 if it's not for an empty file / compressed empty file
+                if elem_md5 != "d41d8cd98f00b204e9800998ecf8427e" and elem_md5 != "7029066c27ac6f5ef18d660d5741979a":
```

Comment:
Here I think it would make sense to test for the size being 0 instead. 

Also, rather than going for checksums for compressed, I'd rather go for trying to decompress small suspectible files and looking at the length of the decompressed data (the 7029066c27ac6f5ef18d660d5741979a seems to be a lucky catch because some library uses very few gzip features, but it will not pickup the general case where we have a timestamp - here that's not very hard to do, though).

Reply:
(Possibly was unclear; I don't know what's used to create these 7029066c27ac6f5ef18d660d5741979a checksum files, but a module that uses e.g. `pigz` for compression typically wouldn't be caught by this, and python seems to have an easy to use gzip module that can be used for very small files to check for empty data.)

Reply:
Thinking about this more - we should never have the more complex case where the md5 varies due to time stamp. We repeat the tests to ensure that the md5 is static - if it includes a time stamp then we can't have it in this file.

So now I'm erring back towards keeping the code simple and just checking for predefined md5s. I guess that there could still be other static checksums from different compression methods of empty files but I'd hope that it's not that many..

Reply:
Could also check for shared md5s across all of the modules. Then these should show up fairly easily I'd hope..

Reply:
Yes, but in the general case, repeated gzip of an empty file won't yield the same checksum. The one seen here is a special sparse one.

And empty outputs imply something being wrong, so I'd encourage catching that.

Reply:
Ok, so as an additional check to catch things that might be going wrong, it's still a good addition 👍🏻 (just not for the QC of "just" the YAML files)

Reply:
Alright, I wrote the code and it was super simple so I'm sold 👍🏻

Reply:
👍
(Generally, it might make sense only to do the test if the size is below a threshold, but this is a good reason to keep down files in tests :)